src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateVirsions = [];
+  const copyState = { ...state };
+
+  for (const action of actions) {
+    Object.assign(copyState, action.extraData);
```

Comment:
you should only do it in `addProperties` action, that's the only actions with `extraData`